### PR TITLE
#45 #47 #48 passport

### PR DESCRIPTION
**Pull request by LChoiSH**
#45 
#47 
#48

### CODE REVIEW

File: Backend/lib/passport.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+module.exports = (app) => {
+
+  const passport = require('passport');
+  LocalStrategy = require('passport-local').Strategy;
+
+  app.use(passport.initialize());
+  app.use(passport.session());
+
+  passport.serializeUser((user, done) => {
+    // console.log('serialize : ', user);
```
Comment (shberry) [inline Backend/lib/passport.js]: remove

File: Backend/lib/passport.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    done(null, user.email);
+  });
+
+  passport.deserializeUser((id, done) => {
+    console.log("deserialize", id);
```
Comment (shberry) [inline Backend/lib/passport.js]: remove if this was just for testing purpose

File: Backend/lib/passport.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  });
+
+  passport.use(new LocalStrategy(
+    {
+      usernameField: 'email',
+      passwordField: 'password',
+      passReqToCallback: true
+    },
+    (req, username, password, done) => {
+      let hashPassword = crypto.createHash("sha512").update(req.query.password).digest("hex");
```
Comment (shberry) [inline Backend/lib/passport.js]: maybe change the variable name to hashedPassword?

File: Backend/lib/passport.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      models.user.findAll({
+        where: {
+          email: req.query.email
+        }
+      }).then(result => {
+        if (result == "") {
```
Comment (shberry) [inline Backend/lib/passport.js]: is there a case where result is null?

File: Backend/router/v1/report.ctrl.js [hunk 5]
```diff
@@ -1,41 +1,70 @@
...
-    time: req.body.data[0].time,
-    who: req.body.data[0].who,
-    details: req.body.data[0].details,
-  }).then((result) => {
-    res.status(201);
-    res.send("good");
-  });
-}
+  if(req.isAuthenticated()){
+    console.log("User id : ", req.user[0].dataValues.id);
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: Please remove all console.log that you used for testing purpose

Comment (shberry) [reply Backend/router/v1/report.ctrl.js]: If you added the log for some other reason, please add a comment for your justification

File: Backend/router/v1/report.ctrl.js [hunk 6]
```diff
@@ -1,41 +1,70 @@
...
+      what: req.body.data[0].what,
+      location: req.body.data[0].location,
+      time: req.body.data[0].time,
+      who: req.body.data[0].who,
+      details: req.body.data[0].details,
+      userID: req.user[0].dataValues.id
+    }).then((result) => {
+      res.status(201);
+      res.json({
+        msg: "make report successfully"
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: Not sure if I understand this message - are you suggesting the frontend to make a successful report? Or did you mean 'report made successfully'?

File: Backend/router/v1/report.ctrl.js [hunk 7]
```diff
@@ -1,41 +1,70 @@
...
+      });
+    });
+  }else{
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: } else {

File: Backend/router/v1/report.ctrl.js [hunk 8]
```diff
@@ -1,41 +1,70 @@
...
+    res.status(403);
+    res.json({
+      msg : "you are not logined"
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: logged in

File: Backend/router/v1/report.ctrl.js [hunk 9]
```diff
@@ -1,41 +1,70 @@
...
-}
+  if(req.isAuthenticated()){
+    models.report.findAll({
+      where: {
+        userID: req.user[0].dataValues.id
+      }
+    }).then((result) => {
+      res.status(200);
+      res.json(result);
+    }).catch((err) => {});
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: should frontend know about this error case? your current logic doesn't do anything.

File: Backend/router/v1/report.ctrl.js [hunk 10]
```diff
@@ -1,41 +1,70 @@
...
-  //When ID is not a number
-  if (Number.isNaN(reportId)) {
-    return res.status(400).end()
-  }
-  models.report.findAll({
-    where: {
-      id: reportId
+  if(req.isAuthenticated()){
+    const reportId = parseInt(req.params.reportId)
+    //When ID is not a number
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: space after //

File: Backend/router/v1/user.js [hunk 11]
```diff
@@ -1,84 +1,79 @@
...
+      });
     }
-  }).catch(err => {
-    res.send("err");
   });
-});
 
-router.get('/signin', (req, res) => {
-  let session = req.session;
+  router.post('/signup', (req, res) => {
```
Comment (shberry) [inline Backend/router/v1/user.js]: why did you add extra indentation? same below

File: Backend/app.js [hunk 12]
```diff
@@ -1,19 +1,29 @@
...
-const router = express.Router();
-const rootRouter = require('./router/index');
 const bodyParser = require('body-parser');
+const router = express.Router();
+const session = require('express-session');
 
 models.sequelize.sync();
 
+app.use(session({
+  secret: 'asadlkj!a@#a!@a#dfgasdg',
```
Comment (mtnviewlab) [inline Backend/app.js]: what is the secret here?

Comment (ethkim) [reply Backend/app.js]: passwords like this should be stored in a separate file (and read in at server loading time), which is excluded from git by .gitignore. :)

File: Backend/router/v1/report.ctrl.js [hunk 13]
```diff
@@ -47,11 +76,47 @@ const deleteReport = (req, res) => {
...
+      });
+    }).catch((err) => {
+      res.json({
+        msg: "update fail"
+      });
+    });
+  } else {
+    res.status(401);
+    res.json({
+      msg: "you are not login"
```
Comment (ethkim) [inline Backend/router/v1/report.ctrl.js]: Use proper English in all debug messages. In fact, this message is a proper json response to the API (i.e. will be exposed to developers if we decide to open the API access), so it's more user-facing than debug messages. 

"You are not logged in."

File: Backend/router/v1/report.js [hunk 14]
```diff
@@ -2,17 +2,20 @@ const express = require('express');
...
+module.exports = () => {
+  router.use((req, res, next) => {
+    next();
+  });
 
-router.all('/', (req, res) => {
-  res.send("this is report root");
-});
+  router.all('/', (req, res) => {
+    res.send("this is report root");
```
Comment (ethkim) [inline Backend/router/v1/report.js]: Fix English. And everywhere in the repository.

File: Backend/lib/passport.js (hunk 4 above)
Comment (LChoiSH) [reply Backend/lib/passport.js]: if DB do not have certain email, it's result is null.

File: Backend/router/v1/report.ctrl.js (hunk 6 above)
Comment (LChoiSH) [reply Backend/router/v1/report.ctrl.js]: the second one. 'report made successfully' and give the result to client.

File: Backend/router/v1/user.js (hunk 11 above)
Comment (LChoiSH) [reply Backend/router/v1/user.js]: I cant understand. Would you explain more exactly?

File: Backend/router/v1/report.js (hunk 14 above)
Comment (LChoiSH) [reply Backend/router/v1/report.js]: @ethkim  I cant understand Fix English. Could you explain more?

Comment (shberry) [reply Backend/router/v1/report.js]: Update the message so people we better understand what it means. When I read 'this is report root', I have no idea what the situation is or what it means.

File: Backend/router/v1/user.js (hunk 11 above)
Comment (shberry) [reply Backend/router/v1/user.js]: Did you look at to see what 'indentation' means? The first sentence in wikipedia gives you enough details to understand this.

File: Backend/router/v1/report.ctrl.js (hunk 9 above)
Comment (shberry) [reply Backend/router/v1/report.ctrl.js]: Then add a TODO for eunyoung and let her delete the TODO when she actually changes it.

File: Backend/router/v1/report.ctrl.js (hunk 6 above)
Comment (shberry) [reply Backend/router/v1/report.ctrl.js]: Then please update. When I ask you a question, I'm not expecting you to add another comment explaining. What I want is for you to update the code so I don't have to ask the question at the first place.

File: Backend/lib/passport.js (hunk 4 above)
Comment (shberry) [reply Backend/lib/passport.js]: Again, when I ask a question, I want you to update the code so I don't have to ask this question. If there is a null case, please take care of it.

**Comment by LChoiSH** [comment]
@shberry @mtnviewlab
Could you approve this?

File: Backend/app.js [hunk 15]
```diff
@@ -1,19 +1,30 @@
...
 }));
 
+const passport = require('./lib/passport')(app);
+const rootRouter = require('./router/index')(passport);
+
 app.use('/api', rootRouter);
 
-app.listen(3000, () => {
+app.listen(80, () => {
   console.log("Server Start");
```
Comment (shberry) [inline Backend/app.js]: Is this something we want to log?

File: Backend/config/session.json [hunk 16]
```diff
@@ -0,0 +1,10 @@
+{
+  "setting": {
+    "secret" : "",
```
Comment (shberry) [inline Backend/config/session.json]: what is 'secret'? Can we come up with a better name?

File: Backend/lib/passport.js [hunk 17]
```diff
@@ -0,0 +1,51 @@
+const models = require('../models');
+const crypto = require('crypto')
+
+module.exports = (app) => {
+
```
Comment (shberry) [inline Backend/lib/passport.js]: remove

File: Backend/router/v1/report.ctrl.js [hunk 18]
```diff
@@ -1,52 +1,73 @@
...
+    res.json({
+      msg: "Can't find this report"
+    });
+  }
+};
 
 const deleteReport = (req, res) => {
   const reportId = parseInt(req.query.reportId);
-  //When ID is not a number
+
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: remove

File: Backend/router/v1/report.ctrl.js [hunk 19]
```diff
@@ -55,51 +76,62 @@ const deleteReport = (req, res) => {
...
+    }).catch((err) => {
+      //TODO(eunyoung): Handling Error messages.
+      res.json({
+        msg: "update fail"
+      });
+    });
+  } else {
+    res.status(401);
+    res.json({
+      msg: "you are not login"
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: logged in

File: Backend/router/v1/report.js [hunk 20]
```diff
@@ -2,18 +2,20 @@ const express = require('express');
...
+module.exports = () => {
+  router.use((req, res, next) => {
+    next();
+  });
 
-router.all('/', (req, res) => {
-  res.send("this is report root");
-});
+  router.all('/', (req, res) => {
+    res.status(404).send("This is not normal route");
```
Comment (shberry) [inline Backend/router/v1/report.js]: Do you think this message will be helpful when the FE sees this?

File: Backend/router/v1/user.js [hunk 21]
```diff
@@ -1,84 +1,82 @@
...
+      }
+    }).then(result => {
+      // If data is not found. The result is null.
+      if (result == "") {
+        models.user.create({
+          email : req.body.email,
+          password : hashedPassword
+        }).then(result => {
+            res.status(201).json({
+              msg: "make id successfully"
```
Comment (shberry) [inline Backend/router/v1/user.js]: created id successfully? 'make id' means you'll be making an id, which is already done at this stage

File: Backend/router/v1/user.js [hunk 22]
```diff
@@ -1,84 +1,82 @@
...
+            });
+        });
       } else {
-        res.json({msg: "Password not Match"});
-      };
-    };
-  }).catch( err => {
-    res.json({msg: "err"});
+        res.status(409).json({
+          msg: "make id fail"
```
Comment (shberry) [inline Backend/router/v1/user.js]: Not a good message. Maybe 'Creating id failed'?

**Comment by LChoiSH** [comment]
When I solve the reviews, frontend changed their code. And I do 'git pull --rebase' and squash my code. As the result, frontend code is put on my commit. Can I solve this problem? @mtnviewlab
